akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowMapAsyncSpec.scala

```diff
@@ -17,8 +17,51 @@ import akka.testkit.TestProbe
 import akka.stream.scaladsl.OperationAttributes.supervisionStrategy
 import akka.stream.Supervision.resumingDecider
 import akka.stream.impl.ReactiveStreamsCompliance
+import scala.util.Try
+import scala.concurrent.ExecutionContext
+import scala.util.Failure
+import scala.util.Success
+import org.scalactic.ConversionCheckedTripleEquals
+
+object FlowMapAsyncSpec {
+  class MapAsyncOne[In, Out](f: In ⇒ Future[Out])(implicit ec: ExecutionContext) extends AsyncStage[In, Out, Try[Out]] {
+    private var elemInFlight: Out = _
+
+    override def onPush(elem: In, ctx: AsyncContext[Out, Try[Out]]) = {
+      val future = f(elem)
+      val cb = ctx.getAsyncCallback()
+      future.onComplete(cb.invoke)
+      ctx.holdUpstream()
+    }
+
+    override def onPull(ctx: AsyncContext[Out, Try[Out]]) =
+      if (elemInFlight != null) {
+        val e = elemInFlight
+        elemInFlight = null.asInstanceOf[Out]
```

Comment:
@hepin1989 If you mean two lines up, then `ne` and `!=` are absolutely identical when involving `null` (well, almost: sometimes the compiler seems to emit `ifnull` and sometimes `ifnonnull` bytecodes, but that should not matter).